src/frequenz/dispatch/_managing_actor.py

```diff
@@ -106,26 +103,45 @@ async def run():
 
     def __init__(
         self,
-        actor: Actor | Set[Actor],
-        dispatch_type: str,
         running_status_receiver: Receiver[Dispatch],
-        updates_sender: Sender[DispatchUpdate] | None = None,
+        actor: Actor | Set[Actor] | None = None,
```

Comment:
I was going for an actor factory. Will submit my WIP as a draft PR so maybe you can get some inspiration about it, or we can discuss pros and cons from the different approaches.